src/components/layer-controller/reducer.js

```diff
@@ -35,16 +35,33 @@ function channelsToLayerProps(channels) {
   };
 }
 
+function getNewChannelProperty(channel, property, value) {
+  if (property === 'visibility') {
+    return !channel.visibility;
+  }
+  if (property === 'selection') {
+    return { ...channel[property], ...value };
+  }
+  return value;
+}
+
 export default function reducer(channels, action) {
   const { type, layerId, payload } = action;
   switch (type) {
-    case 'CHANGE_PROPERTY': {
+    case 'CHANGE_SINGLE_CHANNEL_PROPERTY': {
+      // property is something like "selection" or "slider."
+      // value is the actual change, like { channel: "DAPI" }.
```

Comment:
Yeah really think dimension indices is the best representation. We can easily index the values of a dimension (by the index) to show the human readable label.